SimTKcommon/src/ParallelExecutor.cpp

```diff
@@ -66,36 +79,48 @@ ParallelExecutorImpl::~ParallelExecutorImpl() {
     pthread_cond_destroy(&waitCondition);
 }
 ParallelExecutorImpl* ParallelExecutorImpl::clone() const {
-    return new ParallelExecutorImpl(threads.size());
+    return new ParallelExecutorImpl(NUM_MAX_THREADS);
 }
 void ParallelExecutorImpl::execute(ParallelExecutor::Task& task, int times) {
-    if (times == 1 || threads.size() == 1) {
-        // Nothing is actually going to get done in parallel, so we might as well
-        // just execute the task directly and save the threading overhead.
-        
-        task.initialize();
-        for (int i = 0; i < times; ++i)
-            task.execute(i);
-        task.finish();
-        return;
-    }
-    
-    // Initialize fields to execute the new task.
-    
-    pthread_mutex_lock(&runLock);
-    currentTask = &task;
-    currentTaskCount = times;
-    waitingThreadCount = 0;
-    for (int i = 0; i < (int) threadInfo.size(); ++i)
-        threadInfo[i]->running = true;
-
-    // Wake up the worker threads and wait until they finish.
-    
-    pthread_cond_broadcast(&runCondition);
-    do {
-        pthread_cond_wait(&waitCondition, &runLock);
-    } while (waitingThreadCount < (int) threads.size());
-    pthread_mutex_unlock(&runLock);
+  if (min(times, NUM_MAX_THREADS) == 1) {
+      //(1) NON-PARALLEL CASE:
+      // Nothing is actually going to get done in parallel, so we might as well
+      // just execute the task directly and save the threading overhead.
+      task.initialize();
+      for (int i = 0; i < times; ++i)
+          task.execute(i);
+      task.finish();
+      return;
+  }else{
```

Comment:
It is actually more clear not to use an `else` here. The previous `if` ends with a `return` so terminates execution of the method. The rest of the code doesn't need to be indented at all. That way the visible structure better reflects the execution path.